joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/AstCreatorBase.scala

```diff
@@ -14,14 +14,14 @@ abstract class AstCreatorBase(filename: String) {
 
   /** Create a global namespace block for the given `filename`
     */
-  def globalNamespaceBlock(): NewNamespaceBlock = {
-    val absPath  = absolutePath(filename)
+  def globalNamespaceBlock(useAbsolutePath: Boolean = true): NewNamespaceBlock = {
```

Comment:
@max-leuthaeuser @johannescoetzee  the other uses of this helper seem to be `javasrc2cpg, `jssrc2cpg` and `c2cpg` - do you see any problem with changing its behavior to use relative paths? (Instead of adding the option for relative paths like this diff is currently doing.)

At the end of the day we shouldn't have absolute paths in those frontends either.

Reply:
`javasrc2cpg` currently uses absolute paths as a default everywhere, so that just needs to be fixed in general. As far as this particular helper goes, however, I don't see any issue with using the relative path by default in javasrc.